### DISCUSSION TITLE
Backup grows with 600 MB everyday, how to diagnose storage issues?

### DISCUSSION THREAD

**Discussion by 98mux** [pocketbase/pocketbase#6023] [Q&A] [answered]
My database suddenly started growing with 600MB everyday, possibly from one of my updates.

I've looked at all of my tables and none of them have grown in a way that would make sense with 600MB growth/day

<img width="807" alt="Screenshot 2024-12-04 at 13 37 07" src="https://github.com/user-attachments/assets/5d6c9a89-be05-46d8-97c7-35e34c4be5dc">

<img width="164" alt="Screenshot 2024-12-04 at 13 38 21" src="https://github.com/user-attachments/assets/73871bd7-1690-4473-a051-7bfc56128051">

My backups have been stable at 400MB and now it's growing with 600MB everyday

Is there any way I can diagnose storage issue? I've looked at all of the tables and looked at all of the JSON fields inside those tables. I've also looked at my recent commits and nothing obvious sticks out

<img width="578" alt="Screenshot 2024-12-04 at 13 56 08" src="https://github.com/user-attachments/assets/b431c21c-6c64-44f6-866c-5d0bd018676c">

<img width="578" alt="Screenshot 2024-12-04 at 13 58 18" src="https://github.com/user-attachments/assets/f3820012-c026-44fd-93e5-85a9b4927958">

I've looked inside the backups ^ The data.db-wal file seems to be the culprit - no idea what it does. On a backup before the growth the data-db-wal was really small and then it became +1 GB a day after as seen on the screenshots above

My pocketbase version is PocketBase v0.22.19

**Comment by ganigeorgiev** [comment, marked as answer]
Upgrade to PocketBase [v0.22.26+](https://github.com/pocketbase/pocketbase/releases/tag/v0.22.26).

**Reply by 98mux** [reply]
Thank you! Will do

**Comment by 98mux** [comment]
Turning the instance off and on helped a lot - and upgrading fixed it too. Now the backup is 300MB 🐈 thanks for the help ganigeorgiev
